File: .scratch/analyze_db.py
import json
import re
from collections import Counter

with open('js/surface-data.js', 'r', encoding='utf-8') as f:
    text = f.read()

# strip JS declaration: const SURFACE_DATA = { ... };
# find first { and last }
start = text.find('{')
end = text.rfind('}')
json_str = text[start:end+1]

data = json.loads(json_str)

devices = data.get('devices', [])
print(f'Total devices in SURFACE_DATA.devices: {len(devices)}')

print('\n=== COUNTS BY LINE ===')
print(Counter(d.get('line') for d in devices))

print('\n=== COUNTS BY SERIES ===')
print(Counter(d.get('series') for d in devices))

print('\n=== CONSUMER DEVICES BY SERIES ===')
for s in ['pro', 'laptop', 'sls', 'book', 'go', 'laptopgo', 'studio', 'duo']:
    s_devs = [d for d in devices if d.get('line') == 'consumer' and d.get('series') == s]
    print(f'-- Series: {s} ({len(s_devs)}) --')
    for d in s_devs:
        print(f"   {d['id']}: {d['name']} | Colors: {d.get('colors')} | Price: {d.get('startingPrice')} | Link: {d.get('sourceUrl')}")

print('\n=== BUSINESS DEVICES BY SERIES ===')
for s in ['pro', 'laptop', 'sls', 'book', 'go', 'laptopgo', 'studio', 'duo']:
    s_devs = [d for d in devices if d.get('line') == 'business' and d.get('series') == s]
    print(f'-- Series: {s} ({len(s_devs)}) --')
    for d in s_devs:
        print(f"   {d['id']}: {d['name']} | Colors: {d.get('colors')} | Price: {d.get('startingPrice')} | Link: {d.get('sourceUrl')}")

with open('.scratch/parsed_devices.json', 'w', encoding='utf-8') as out:
    json.dump(devices, out, ensure_ascii=False, indent=2)

File: .scratch/analyze_db.js
const fs = require('fs');
const SURFACE_DATA = require('../js/surface-data.js');

console.log('SURFACE_DATA loaded successfully!');
console.log('Devices count:', SURFACE_DATA.devices.length);
console.log('Categories count:', SURFACE_DATA.categories.length);

const lines = {};
const series = {};
SURFACE_DATA.devices.forEach(d => {
  lines[d.line] = (lines[d.line] || 0) + 1;
  series[d.series] = (series[d.series] || 0) + 1;
});
console.log('Lines:', JSON.stringify(lines, null, 2));
console.log('Series:', JSON.stringify(series, null, 2));

const report = {
  consumer: {},
  business: {}
};

SURFACE_DATA.devices.forEach(d => {
  const line = d.line || 'unknown';
  const ser = d.series || 'unknown';
  if (!report[line]) report[line] = {};
  if (!report[line][ser]) report[line][ser] = [];
  report[line][ser].push({
    id: d.id,
    name: d.name,
    generation: d.generation,
    subSeries: d.subSeries,
    colors: d.colors,
    chips: d.chips,
    startingPrice: d.startingPrice,
    sourceUrl: d.sourceUrl
  });
});

fs.writeFileSync('.scratch/db_summary.json', JSON.stringify(report, null, 2), 'utf-8');
console.log('Written .scratch/db_summary.json');
